### PR TITLE
feat(linter): eslint-plugin-vitest/no-import-node-test

### PR DESCRIPTION
**Pull request by eryue0220**
support [eslint-plugin-vitest/no-import-node-test](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/no-import-node-test.md)

### CODE REVIEW

**Review by mysteryven — APPROVED** [review]

File: crates/oxc_linter/src/rules/vitest/no_import_node_test.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::ImportDeclaration(import_decl) = node.kind() {
+            if import_decl.source.value.eq("node:test") {
+                let span = import_decl.source.span;
+
+                ctx.diagnostic_with_fix(no_import_node_test(span), |fixer| {
+                    fixer.replace(span, "\"vitest\"")
+                });
+            }
+        };
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/vitest/no_import_node_test.rs:57]: Let's try use [module record](https://github.com/oxc-project/oxc/blob/a207923af1b87953e89ecbf0fe272cf7134b71ef/crates/oxc_syntax/src/module_record.rs#L17)

**Review by mysteryven — CHANGES_REQUESTED** [review]
Let's try module record to see if it can improve our performance👀

To run benchmark for vitest rule, we also need to modify: https://github.com/oxc-project/oxc/blob/7a75e0f8a77006ef32fa0db45a329d961e70bdf3/tasks/benchmark/benches/linter.rs#L44-L51

**Comment by eryue0220** [comment]
> Let's try module record to see if it can improve our performance👀
> 
> To run benchmark for vitest rule, we also need to modify:
> 
> https://github.com/oxc-project/oxc/blob/7a75e0f8a77006ef32fa0db45a329d961e70bdf3/tasks/benchmark/benches/linter.rs#L44-L51

still the same problem. :(

**Comment by mysteryven** [comment]
> still the same problem. :(

We can try to use module record. e.g. 

1. https://github.com/oxc-project/oxc/blob/7a75e0f8a77006ef32fa0db45a329d961e70bdf3/crates/oxc_linter/src/rules/import/no_self_import.rs#L30-L46
2. https://github.com/oxc-project/oxc/blob/7a75e0f8a77006ef32fa0db45a329d961e70bdf3/crates/oxc_linter/src/frameworks.rs#L83-L85

**Comment by eryue0220** [comment]
Thanks, it works!!

**Review by mysteryven — APPROVED** [review]
Nice!

**Pull request merged**